### PR TITLE
Bump Julia to 0.6, fix 0.7 breakage

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: src/CoverageBase.jl [hunk 1]
```diff
@@ -5,16 +5,22 @@ export testnames, runtests
 
 const need_inlining = []
 
+if VERSION < v"0.7.0-DEV.3073"
```
Comment (kshyatt) [inline src/CoverageBase.jl]: why not use Compat's binding for this?

Comment (ararslan) [reply src/CoverageBase.jl]: Seemed unnecessary to add a dependency on Compat for that one thing, but I guess we have an indirect dependency on it from Coverage so I should just do that.

**Review by kshyatt — APPROVED** [review]

**Comment by ararslan** [comment]
The CI failures will hopefully be fixed by https://github.com/JuliaCI/Coverage.jl/pull/146.

**Comment by ararslan** [comment]
Passing tests, the failure was because we were submitting coverage to Coveralls but hadn't actually turned on Coveralls for this repo. I set that up before the most recent close/reopen.

**Pull request merged**